types/src/unit_tests/canonical_serialization_examples.rs

```diff
@@ -0,0 +1,319 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This test verifies the behavior of the Canonical Serializer against the test vectors
+//! provided in the Canonical Serializer document.
+//! See: common/canonical_serializer/README.md
+
+use crate::{
+    access_path::AccessPath,
+    account_address::AccountAddress,
+    byte_array::ByteArray,
+    transaction::{Program, RawTransaction, TransactionArgument, TransactionPayload},
+    write_set::{WriteOp, WriteSet, WriteSetMut},
+};
+use canonical_serialization::SimpleSerializer;
+use std::time::Duration;
+
+#[test]
+fn test_access_path_canonical_serialization_example() {
```

Comment:
Can you include any code / scripts you used to generate these as comments?

Reply:
Good point. I should have done this a bit more formulaic... I wrote a bunch of hacky scripts to do it. Will attach slightly cleaned up format to get your feedback. Perhaps we should invest in a test vector generator.